tutorials/tree/dictionary/writeTree.cpp

```diff
@@ -0,0 +1,61 @@
+// ##########################################################################
+// # Alvaro Tolosa Delgado @ IFIC(Valencia,Spain)  alvaro.tolosa@ific.uv.es #
+// # Copyright (c) 2018 Alvaro Tolosa. All rights reserved.		 #
+// ##########################################################################
+
+#ifndef __writeTree_cpp__
+#define __writeTree_cpp__
+
+#include <iostream>
+
+#include <TFile.h>
+#include <TTree.h>
+#include <TTreeReader.h>
+#include <TTreeReaderValue.h>
+
+#include "data2Tree.hpp"
+
+void writeTree()
+{
+
+   TFile *ofile = TFile::Open("testFile.root", "recreate");
+   if (!ofile) {
+      std::cerr << " File not found or already exists" << std::endl;
+      return;
+   }
+   TTree *myTree = new TTree("myTree", "");
```

Comment:
Same here for the `std::unique_ptr`.

Reply:
ROOT segfaults if I replace
`TTree *myTree = new TTree("myTree", "");`
by
`std::unique_ptr<TTree> myTree = std::make_unique<TTree>("myTree", "");`

This is the backtrace:
```
 *** Break *** segmentation violation



===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
#0  0x00007fd5aea6a5db in waitpid () from /lib64/libc.so.6
#1  0x00007fd5ae9ccce7 in do_system () from /lib64/libc.so.6
#2  0x00007fd5b3725563 in TUnixSystem::StackTrace() () from /usr/local/root_6.26.06/lib/libCore.so
#3  0x00007fd5b3727f95 in TUnixSystem::DispatchSignals(ESignals) () from /usr/local/root_6.26.06/lib/libCore.so
#4  <signal handler called>
#5  0x000000000040465a in std::default_delete<TTree>::operator()(TTree*) const ()
#6  0x0000000000404317 in std::unique_ptr<TTree, std::default_delete<TTree> >::~unique_ptr() ()
#7  0x0000000000403e45 in writeTree() ()
#8  0x0000000000402e77 in main ()
===========================================================


The lines below might hint at the cause of the crash.
You may get help by asking at the ROOT forum https://root.cern/forum
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#5  0x000000000040465a in std::default_delete<TTree>::operator()(TTree*) const ()
#6  0x0000000000404317 in std::unique_ptr<TTree, std::default_delete<TTree> >::~unique_ptr() ()
#7  0x0000000000403e45 in writeTree() ()
#8  0x0000000000402e77 in main ()
===========================================================
```

Reply:
This technically because of the:
```
ofile->Close();
```
which deletes the `TTree` object.  This call to `Close` is unnecessary as it is called by the `TFile` destructor and should be replacement by a call to `ofile->Write()` (which makes the call to `tree->Write();` redundant)

Reply:
great, at the end I have changed this part, thanks!